src/commons/withScrollEnabler.tsx

```diff
@@ -0,0 +1,79 @@
+import _ from 'lodash';
+import React, {useState, useEffect, useCallback, useRef} from 'react';
+// eslint-disable-next-line no-unused-vars
+import {FlatListProps, ScrollViewProps} from 'react-native';
+import {Constants} from 'react-native-ui-lib';
+import forwardRef from './forwardRef';
+
+export type WithScrollEnablerProps = FlatListProps<any> | ScrollViewProps;
+function withScrollEnabler<PROPS extends WithScrollEnablerProps>(WrappedComponent: React.ComponentType<WithScrollEnablerProps>): React.ComponentType<PROPS> {
+  const ScrollEnabler = (props: WithScrollEnablerProps) => {
+    const [scrollEnabled, setScrollEnabled] = useState(true);
+    const [contentSize, setContentSize] = useState(0);
+    const [layoutSize, setLayoutSize] = useState(0);
+
+    const scrollViewRef = useRef<WithScrollEnablerProps>(null);
+
+    useEffect(() => {
```

Comment:
Why do u need an access to the Scroll element ref?

Reply:
Moved to screen